examples/src/main/java/tachyon/examples/Performance.java

```diff
@@ -228,18 +227,18 @@ public TachyonReadWorker(int id, int left, int right, ByteBuffer buf) throws IOE
     }
 
     public void readPartition() throws IOException {
-      TachyonByteBuffer buf;
       if (sDebugMode) {
+        ByteBuffer buf = ByteBuffer.allocateDirect((int) sBlockSizeBytes);
         LOG.info("Verifying the reading data...");
 
         for (int pId = mLeft; pId < mRight; pId ++) {
           TachyonFile file = mTC.getFile(new TachyonURI(sFileName + (pId + sBaseFileNumber)));
-          buf = file.readByteBuffer(0);
-          IntBuffer intBuf;
-          intBuf = buf.mData.order(ByteOrder.nativeOrder()).asIntBuffer();
+          InputStream is = file.getInStream(ReadType.CACHE);
+          is.read(buf.array());
```

Comment:
buf is a direct buffer, so it is probably not backed by an array? Doesn't hasArray() have to be called? Maybe buf should just go back to a normal ByteBuffer?